packages/fiber/src/core/store.ts

```diff
@@ -217,6 +214,22 @@ const createStore = (
     const setPerformanceCurrent = (current: number) =>
       set((state) => ({ performance: { ...state.performance, current } }))
 
+    const handleXRFrame = (timestamp: number) => {
+      const state = get()
+      if (state.frameloop === 'never') return
+
+      advance(timestamp, true)
+    }
+
+    const handleSessionChange = () => {
+      gl.xr.enabled = gl.xr.isPresenting
+      gl.setAnimationLoop(gl.xr.isPresenting ? handleXRFrame : null)
+    }
+
+    // Update render mode on session change
+    gl.xr.addEventListener('sessionstart', handleSessionChange)
+    gl.xr.addEventListener('sessionend', handleSessionChange)
```

Comment:
How can I have Zustand listen to `state.gl.xr.isPresenting` instead?

Reply:
as in you want to subscribe to an object property as opposed to to the whole object?

Reply:
Yes, although `gl.xr` can be undefined if `gl` is not a WebGLRenderer.

Reply:
I've asked Daiti in discord. Will report back.

Reply:
yeah so we do:

```js
useStore.subscribe(callbackFn, state => state.foo.bar)
```

And since `xr` can be `undefined` we can just use `?.`

Reply:
I see, thanks. I'm curious as to why that's only firing on re-render with Vite's HMR.

Reply:
I can take a look if you'd like?

Reply:
I appreciate the help.

Just committed an example case with instructions on how to recreate this behavior. The button sets `gl.xr.isPresenting` in a useEffect to emulate WebXRManager's session behavior.

https://user-images.githubusercontent.com/23324155/135752669-f589f37b-7fea-472d-8090-afd8442ec493.mp4

Reply:
@CodyJasonBennett  you can't observe internals. how could zustand know that threejs has set some deeply nested flag, it can only know about changes because you or someone `set`s them.

Reply:
Thanks. I'll have to defer to listening to session events then.